chcemvediet/apps/inforequests/admin.py

```diff
@@ -3,16 +3,17 @@
 import datetime
 
 from django.contrib import admin
+from django.contrib.admin.actions import delete_selected
```

Comment:
@martinmacko47 Este som zabudol spomentut `snooze`. Ze ci chceme umoznit adminovi tuto moznost, Asi by som to riesil tak, ze cez jeden checkbox, predlzim `snooze` na vsetkych akciach ktore splnaju podmienku ako pri `delete_model`. Alebo to pri bulk delete nemuoznime vobec?

Reply:
Good point. Ano, ak mazeme aspon jednu `EXPIRATION` alebo `APPEAL_EXPIRATION` akciu, ktora ma previous action, pri ktorej sa upravit snooze, tak by sme checkbox zobrazit mali. Ak nemazeme ziadnu taku akciu, checkbox by sa nemal zobrazovat.

Text pri checkboxe by mal byt zhruba: "Extend snooze of previous actions to the following deleted actions by X days from today:", a potom vymenovat akcie, ktorych sa to tyka. Staci jeden checkbox, nemusi byt zvlast pre kazdu akciu.

Btw, bolo by fajn logiku `obj.type in [Action.TYPES.EXPIRATION, Action.TYPES.APPEAL_EXPIRATION] and obj.previous_action` vysunut do mocnej funkcie, nech ju nemame v kode 4x (1x v `delete_model`, 1x v `delete_selected` a 2x v templatoch.)

Reply:
@martinmacko47  "... pri ktorej sa moze upravit snooze". Doteraz sme neoverovali typ predoslej akcie, iba ci existuje. Mam teda doplnit o tuto kontrolu? Ak ano, u ktorych akciach sa moze upravit snooze? V help texte filedu `snooze` je "Ignored for applicant deadlines.", ale v kode je snooze povoleny  aj pre akcie typu Applicant. 
```
# Applicant actions
        if self.type == self.TYPES.REQUEST:
            return Deadline(Deadline.TYPES.OBLIGEE_DEADLINE,
                    self.delivered_date or workdays.advance(self.sent_date, 1),
                    8, Deadline.UNITS.WORKDAYS, self.snooze)

        elif self.type == self.TYPES.CLARIFICATION_RESPONSE:
            return Deadline(Deadline.TYPES.OBLIGEE_DEADLINE,
                    self.delivered_date or workdays.advance(self.sent_date, 1),
                    8, Deadline.UNITS.WORKDAYS, self.snooze)

        elif self.type == self.TYPES.APPEAL:
            return Deadline(Deadline.TYPES.OBLIGEE_DEADLINE,
                    self.delivered_date or workdays.advance(self.sent_date, 6),
                    15, Deadline.UNITS.CALENDAR_DAYS, self.snooze)
```

Vlastne v if podmienke mozeme vyhodit cast `self.can_snooze(obj)`, pretoze to kontrolujeme v `render_delete_form`. Ak je to false, potom sa nevyrendruje do template checkbox a `request.POST.get(u'snooze')` bude tiez False.
```
    def delete_model(self, request, obj):
        ...
        if request.POST:
            if request.POST.get(u'snooze') and self.can_snooze(obj):
```

Reply:
@martinmacko47 A na druhe zamyslenie: V podstate nepotrebujeme ani kontrolu `if request.POST`. `delete_model` metoda je volana v `...admin/options.py` kde uz tato kontrola nastala. Takze by som to cele vymazal. A vlastne to nekontrolujeme ani v metode `DeleteNestedInforequestEmailAdminMixin.render_delete_form`.

Reply:
>  "... pri ktorej sa moze upravit snooze". Doteraz sme neoverovali typ predoslej akcie, iba ci existuje. Mam teda doplnit o tuto kontrolu? Ak ano, u ktorych akciach sa moze upravit snooze?

Tu som sa zle vyjadril. Mal som na mysli "ak mazeme aspon jednu `EXPIRATION` alebo `APPEAL_EXPIRATION` akciu, ktora ma previous action" bez dovetku "pri ktorej sa moze upravit snooze". Aktualne povazujeve vsetky akcie, ktore predchadzaju `EXPIRATION` alebo `APPEAL_EXPIRATION` za take, ktorym mozeme upravit snooze.

Totiz `Branch.add_expiration_if_expired` prida `EXPIRATION` resp. `APPEAL_EXPIRATION` iba ak `Branch.last_action.has_obligee_deadline_missed`. Cize `EXPIRATION` resp. `APPEAL_EXPIRATION` moze nasledovat iba akcie, ktore maju obligee deadline, pri ktorych sa snooze berie do uvahy vzdy.

Jedine, ze by admin rucne zmazal tu predoslu akciu, cim by sa novou predoslou akciou mohlo stat cokolvek. Na to vsak bol upozorneny pri mazani.

> Vlastne v if podmienke mozeme vyhodit cast `self.can_snooze(obj)`, pretoze to kontrolujeme v `render_delete_form`. Ak je to false, potom sa nevyrendruje do template checkbox a `request.POST.get(u'snooze')` bude tiez False.

To v ziadnom pripade, ved medzi `render_delete_form` a `delete_model` sa dostene k slovu FE. Data poslane do `delete_model` posiela FE, tym predsa nemozeme verit a musime ich validovat. Kludne nam FE mohol pozlat parameter `snooze=true` aj pre objety, pre ktore `self.can_snooze(obj)` neplati. Staci, ze niekde ten formular upravil v developer tools v browsri.

**Nikdy** never, ze FE posle iba data, ktore mu do formularu vlozis. BE si musi vsetko znovu zvalidovat.

> V podstate nepotrebujeme ani kontrolu `if request.POST`. `delete_model` metoda je volana v `...admin/options.py` kde uz tato kontrola nastala.

Jj, podmienku `if request.POST` mozeme vyhodit. Ta bude stale true.

Reply:
:ok: